### PR TITLE
User can schedule goal for the whole year

### PR DESCRIPTION
**Pull request by Troff8**
## PR includes

- [x] Feature

## Related issues

Resolve [#519](https://github.com/taskany-inc/issues/issues/519)

## QA Instructions, Screenshots, Recordings
  
image

### CODE REVIEW

File: src/components/EstimateComboBox.tsx [hunk 1]
```diff
@@ -37,10 +37,15 @@ interface EstimateComboBoxProps {
     onChange?: (estimate?: { date: string; q: string; y: string }) => void;
 }
 
-const StyledInput = styled(Input)`
-    min-width: 100px;
+const StyledInput = styled(Input)<{ width?: number }>`
```
Comment (awinogradov) [inline src/components/EstimateComboBox.tsx:40]: Input has prop `brick`, u can use it like `<Input brick="right" ... />`. Button has same prop ;)

File: src/components/EstimateComboBox.tsx [hunk 2]
```diff
@@ -243,7 +283,29 @@ export const EstimateComboBox = React.forwardRef<HTMLDivElement, EstimateComboBo
                     />
                 )}
                 renderItems={(children) => (
-                    <StyledButtonsContainer>{children as React.ReactNode}</StyledButtonsContainer>
+                    <>
+                        <StyledButtonsContainer>{children as React.ReactNode}</StyledButtonsContainer>
+                        <StyledItemsYearContainer>
+                            <InputMask
+                                mask={'9999'} // как по другому можно сделать? если папки .i18n нет
```
Comment (awinogradov) [inline src/components/EstimateComboBox.tsx:290]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number

**Review by awinogradov — CHANGES_REQUESTED** [review]
So, we have Estimate as object `{ q?: string; y: number; date: string }`. In case of whole year estimate for goal we need to have value `{ q: undefined, y: 2023, date: undefined }`. U don't need to calculate quarter of end of the year.

File: src/components/EstimateComboBox.tsx [hunk 3]
```diff
@@ -139,53 +144,72 @@ export const EstimateComboBox = React.forwardRef<HTMLDivElement, EstimateComboBo
...
                 const v = createValue(inputState, locale);
                 setButtonText(formatEstimate(v, locale));
-
                 onChange?.(v);
             }
-        }, [changed, selectedQ, inputState, locale, onChange]);
+            if (selectedQ === undefined) {
+                onChange?.({ q: undefined, y: currentYear, date: undefined });
+            }
+        }, [changed, selectedQ, inputState, locale, onChange, currentYear, quarterInfo.Q4.date]);
```
Comment (awinogradov) [inline src/components/EstimateComboBox.tsx:195]: `quarterInfo.Q4.date` is not used in hook

File: src/components/EstimateComboBox.tsx [hunk 4]
```diff
@@ -139,53 +144,72 @@ export const EstimateComboBox = React.forwardRef<HTMLDivElement, EstimateComboBo
...
+                    setSelectedQ(quarterFromDate(parseLocaleDate(inputState, { locale })));
+                }
+            },
+            [inputState, locale],
+        );
 
         useEffect(() => {
-            if (isValidDate(inputState)) {
-                setSelectedQ(quarterFromDate(parseLocaleDate(inputState, { locale })));
+            if (Object.keys(quarterInfo).length && selectedQ !== undefined) {
```
Comment (awinogradov) [inline src/components/EstimateComboBox.tsx:174]: What the case to have 0 in `Object.keys(quarterInfo).length`?

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/components/EstimateComboBox.tsx (hunk 4 above)
Comment (Troff8) [reply src/components/EstimateComboBox.tsx:174]: at the moment this case can no longer be, I will remove it

**Review by awinogradov — APPROVED** [review]

**Pull request merged**